prov/usnic/src/usdf_eq.c

```diff
@@ -194,11 +197,27 @@ static ssize_t usdf_eq_readerr(struct fid_eq *feq,
 		goto done;
 	}
 
-	ret = usdf_eq_read_event(eq, NULL, entry, sizeof(*entry), flags);
+	ret = usdf_eq_read_event(eq, NULL, &entry, sizeof(entry), flags);
+
+	/* read the user's setting for err_data. */
+	err_data = buffer->err_data;
+	err_data_size = buffer->err_data_size;
+
+	/* Copy the entry. */
+	*buffer = entry;
+
+	/* If version >= 1.5, we copy err_data to user's provided buffer and
+	 * since we overwritten the value above, we restore saved value here. */
+	api_version = eq->eq_fabric->fab_attr.api_version;
+	if (FI_VERSION_GE(api_version, FI_VERSION(1,5))) {
+		buffer->err_data = err_data;
+		buffer->err_data_size = MIN(err_data_size, entry.err_data_size);
+		memcpy(buffer->err_data, entry.err_data, buffer->err_data_size);
+	}
 
 	/* Mark as seen so it can be cleaned on the next iteration of read. */
-	if (entry->err_data_size) {
-		err_data_entry = container_of(entry->err_data,
+	if (entry.err_data_size) {
+		err_data_entry = container_of(&entry.err_data,
```

Comment:
Did you test this? I'm not 100% sure on it, but I feel like that shold be `entry.err_data` not `&entry.err_data`.